src/Microsoft.AspNetCore.Mvc.RazorPages/Infrastructure/PageActionDescriptorProvider.cs

```diff
@@ -57,9 +58,29 @@ protected IList<PageRouteModel> BuildModel()
                 _routeModelProviders[i].OnProvidersExecuted(context);
             }
 
+            RemoveSupersededFallbackRoutes(context);
+
             return context.RouteModels;
         }
 
+        private void RemoveSupersededFallbackRoutes(PageRouteModelProviderContext context)
+        {
+            for (var i = 0; i < context.RouteModels.Count; i++)
+            {
+                var fallbackRouteModel = context.RouteModels[i];
+                if (!fallbackRouteModel.Properties.ContainsKey(FallbackRazorPagesConventions.FallbackPropertyKey))
```

Comment:
Any objection to just making this a first-class part of the model? It's a feature we are adding. I think it deserves it.